sources/osgUtil/Composer.js

```diff
@@ -521,16 +521,27 @@ Composer.Filter.PingPong.prototype = MACROUTILS.objectInherit( Composer.Filter.C
         st0.setAttributeAndModes( this._program );
         st1.setAttributeAndModes( this._program );
 
-        var uniforms = this._uniforms;
+        var k, l, keys, unif, uniforms = this.getStateSet().getUniformList();
         if ( uniforms ) {
-            var keys = window.Object.keys( uniforms );
-            for ( var k = 0, l = keys.length; k < l; k++ ) {
-                var unif = uniforms[ keys[ k ] ];
+            keys = window.Object.keys( uniforms );
+            for ( k = 0, l = keys.length; k < l; k++ ) {
+                unif = uniforms[ keys[ k ] ];
                 st0.addUniform( unif );
                 st1.addUniform( unif );
             }
         }
 
+        uniforms = this._uniforms;
+        if ( uniforms ) {
+            keys = window.Object.keys( uniforms );
```

Comment:
done, starting at line 524 above

```
// PingPong filter is a peculiar Filter where user provides the Camera
// instead of compose::build creating them, and allowing user to provide
// them in the ctor
// To make sure we don't forget any uniform
// we make sure to get uniform from the filter itself and the uniform
// from the parameters
```